### PR TITLE
[CAMEL-19129] Improve CXF SOAP documentation

### PR DESCRIPTION
**Pull request by llowinge**

### CODE REVIEW

**Comment by llowinge** [comment]
@ffang Can you please review ?

**Comment by llowinge** [comment]
Missing parts:
* What to do with the Spring example ? (The Spring details may confuse users who do not use Spring. There should be a separate place for Spring specific details.) - more info in Jira CAMEL-19129
* Having some best practice security section

I'm not sure, if the "simplest" scenario which i've added to the top is acceptable.

File: components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc [hunk 1]
```diff
@@ -120,921 +120,919 @@ exchange property, `CamelCXFDataFormat`. The exchange key constant is
...
-                }
-            });
         }
-    };
+    }
 }
 ----
 
-== How to get and set SOAP headers in POJO mode
+*Message Mode:* Attachments are not supported as it does not process the
```
Comment (ffang) [inline components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc:677]: Should change "Message Mode" to "RAW Mode"

File: components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc [hunk 2]
```diff
@@ -120,921 +120,919 @@ exchange property, `CamelCXFDataFormat`. The exchange key constant is
...
+-------------------------------------------------------------------------------------------------------
 
-[source,java]
-----
-from(routerEndpointURI).setFaultBody(constant(SOAP_FAULT));
-----
+Take a look at the tests that show how you'd be able to relay/drop
+headers here:
 
+https://github.com/apache/camel/blob/main/components/camel-cxf/src/test/java/org/apache/camel/component/cxf/soap/headers/CxfMessageHeadersRelayTest.java[https://github.com/apache/camel/blob/main/components/camel-cxf/src/test/java/org/apache/camel/component/cxf/soap/headers/CxfMessageHeadersRelayTest.java]
```
Comment (ffang) [inline components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc:831]: Should change to
https://github.com/apache/camel/blob/main/components/camel-cxf/camel-cxf-spring-soap/src/test/java/org/apache/camel/component/cxf/soap/headers/CxfMessageHeadersRelayTest.java

File: components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc [hunk 3]
```diff
@@ -120,921 +120,919 @@ exchange property, `CamelCXFDataFormat`. The exchange key constant is
...
+`<?xml version="1.0" encoding="utf-8"?>`:
+
+---------------------------------------------------------------------------------------
+Error:sendSms: SoapFault exception: [Client] looks like we got no XML document in [...]
+---------------------------------------------------------------------------------------
 
-You can produce a Camel message with attachment to send to a CXF
-endpoint in Payload mode.
+To resolve this issue, you just need to tell StaxOutInterceptor to
+write the XML start document for you, as in the https://github.com/apache/camel/blob/main/components/camel-cxf/src/test/java/org/apache/camel/component/cxf/WriteXmlDeclarationInterceptor.java[WriteXmlDeclarationInterceptor] below:
```
Comment (ffang) [inline components/camel-cxf/camel-cxf-soap/src/main/docs/cxf-component.adoc:929]: Should change to 
https://github.com/apache/camel/blob/main/components/camel-cxf/camel-cxf-soap/src/test/java/org/apache/camel/component/cxf/jaxws/WriteXmlDeclarationInterceptor.java

**Comment by llowinge** [comment]
@ffang Thank you for the review. What do you think about links which are not pointing to `main` branch - eg. "https://github.com/apache/camel/blob/e818e0103490a106fa1538219f91a732ddebc562..." shouldn't i switch them to point to the `main` branch ?

**Comment by ffang** [comment]
> @ffang Thank you for the review. What do you think about links which are not pointing to `main` branch - eg. "https://github.com/apache/camel/blob/e818e0103490a106fa1538219f91a732ddebc562..." shouldn't i switch them to point to the `main` branch ?

Yes, It's better to switch them to point `main` branch

**Comment by llowinge** [comment]
@ffang I think it is OK now. WDYT ?

**Review by ffang — APPROVED** [review]

**Comment by ffang** [comment]
Thanks @llowinge !

**Comment by llowinge** [comment]
@oscerd Could you merge please ?

**Pull request merged**